scripts/perftune.py

```diff
@@ -332,9 +333,20 @@ def auto_detect_irq_mask(cpu_mask, cores_per_irq_core):
 
         return run_hwloc_calc(['--restrict', cpu_mask] + hwloc_args)
 
+# @param headers dict of k:v
+def curl(url, headers=None, method=None, byte=False, timeout=0):
+    req = urllib.request.Request(url, headers=headers or {}, method=method)
+    with urllib.request.urlopen(req, timeout=timeout) as res:
+        if byte:
+            return res.read()
+        else:
+            return res.read().decode('utf-8')
 
 ################################################################################
 class PerfTunerBase(metaclass=abc.ABCMeta):
+    EC2_METADATA_BASE_URL = "http://169.254.169.254/latest/"
+    EC2_METADATA_TOKEN_TTL = 21600
+
```

Comment:
Please, move these into a method that uses it or make these a "property".

Reply:
Moved ttl to `__metadata_token(self)`, and make base_url to a property.